test-framework/src/main/java/org/elasticsearch/test/ExternalNode.java

```diff
@@ -233,7 +229,11 @@ synchronized boolean running() {
 
     @Override
     public void close() {
-        stop();
+        try {
+            stop();
+        } catch (InterruptedException e) {
```

Comment:
`#close()' could either be invoked directly or from within a try-with-resources block and I did not come across either.

Unfortunately, bubbling up is not an option as we implement `Closeable` from the JDK. Hence, I sensed restoring the flag is the most sensible option in this case.